### PR TITLE
Add support OpenBSD platforms.

### PR DESCRIPTION
**Pull request by fnichol**
This commit adds support for OpenBSD (tested with OpenBSD 5.6/amd64).
There were 3 issues that needed to be addressed to make this possible:
- The `configure` script uses a shell array which isn't supported in
  bourne shell or ksh (at least under OpenBSD), but is supported in
  bash. Invoking `configure` with bash on OpenBSD resolved this
  failure.
- OpenBSD wasn't in the supported `host_os` detection logic, but falls
  under the FreeBSD case, which accomplished by adding
  `--with-host-os=linux` to the `configure` invocation. I know, totally
  makes sense to me too.
- There appears to be GNUisms in the Makefile code, thus using `gmake`
  on OpenBSD helps the build complete.

Consequently for OpenBSD platforms, the following packages must be
installed:
- comp${OSrev}.tgz (the compiler toolchain system component)
- bash
- gmake

### CODE REVIEW

**Comment by joemiller** [comment]
also need autoconf pkg installed. tested with `autoconf-2.69p1` since there are several choices on openbsd 5.7.

**Comment by joemiller** [comment]
it would be awesome to get this merged. it is necessary to do chef omnibus builds on openbsd.

File: ext/libgecode3/extconf.rb [hunk 1]
```diff
@@ -25,9 +29,17 @@ def self.prefix
     PREFIX
   end
 
+  def self.sh
+    if openbsd?
+      "bash"
+    else
+      "sh"
+    end
+  end
```
Comment (lamont-granquist) [inline ext/libgecode3/extconf.rb]: this should probably generically try "bash" if its in the path (with a little help to make sure /usr/local/bin is in the path) and then resolve to the full path of bash, and then fallback to /bin/sh

File: ext/libgecode3/extconf.rb [hunk 2]
```diff
@@ -60,6 +73,10 @@ def self.setup_env
       ENV['CXX'] = 'g++44'
     end
 
+    if openbsd?
+      ENV['CONFIG_SHELL'] = '/usr/local/bin/bash'
+    end
```
Comment (lamont-granquist) [inline ext/libgecode3/extconf.rb]: this should get set to whatever shell we picked above.

File: ext/libgecode3/extconf.rb [hunk 3]
```diff
@@ -81,6 +98,14 @@ def self.patch_configure
     end
   end
 
+  def self.make
+    if openbsd?
+      "gmake"
+    else
+      "make"
+    end
+  end
```
Comment (lamont-granquist) [inline ext/libgecode3/extconf.rb]: this should probably be changed to prefer gmake over make if its found in the path.

File: ext/libgecode3/extconf.rb [hunk 4]
```diff
@@ -40,6 +52,7 @@ def self.configure_cmd
       --disable-flatzinc
     ]
     args << "--with-host-os=windows" if windows?
+    args << "--with-host-os=linux" if openbsd?
```
Comment (lamont-granquist) [inline ext/libgecode3/extconf.rb]: should this be all BSDs?  (plus SunOS per another PR)

**Comment by tas50** [comment]
@fnichol Any interest in updating this per the review comments?

**Comment by tas50** [comment]
Closing this out since we never heard back

**Pull request closed without merging**